mcp/mastra/verticals/weather/shortcuts.ts

```diff
@@ -0,0 +1,114 @@
+import { z } from 'zod';
+import { createTool } from '../../utils/tool-factory.js';
+import { inferUserLocation } from '../internet-of-things/tools.js';
+
+/**
+ * Shortcuts are tools that piggy-back on other verticals' capabilities.
+ * They allow a vertical to leverage tools from other domains while maintaining
+ * its own domain-specific interface.
+ */
+
+interface UserLocationInfo {
+  userId: string;
+  userName: string;
+  state: string;
+  latitude: number | null;
+  longitude: number | null;
+  gpsAccuracy: number | null;
+  lastChanged: string;
+  source: string;
+  distancesFromZones: Array<{
+    zoneName: string;
+    zoneId: string;
+    distanceMeters: number | null;
+    isInZone: boolean;
+  }>;
+}
+
+/**
+ * Get the user's current location for weather-related queries.
+ * This shortcut uses the IoT vertical's inferUserLocation tool to determine
+ * where the user is located, enabling location-aware weather features.
+ */
+export const getUserCurrentLocation = createTool({
+  id: 'getUserCurrentLocation',
+  description:
+    "Get the current location of a user for weather purposes. Uses IoT device tracking to determine the user's GPS coordinates and zone information.",
+  inputSchema: z.object({
```

Comment:
Shortcuts should always reuse the input and output schema of the tool they call. Make a new createShortcut utility function that's used for all shortcuts to help enforce this.